src/data/mv_polynomial/basic.lean

```diff
@@ -140,14 +140,30 @@ by simp [C, monomial, single_mul_single]
 @[simp] lemma C_pow (a : α) (n : ℕ) : (C (a^n) : mv_polynomial σ α) = (C a)^n :=
 by induction n; simp [pow_succ, *]
 
-lemma C_injective (σ : Type*) (R : Type*) [comm_ring R] :
+lemma C_injective (σ : Type*) (R : Type*) [comm_semiring R] :
   function.injective (C : R → mv_polynomial σ R) :=
 finsupp.single_injective _
 
-@[simp] lemma C_inj {σ : Type*} (R : Type*) [comm_ring R] (r s : R) :
+@[simp] lemma C_inj {σ : Type*} (R : Type*) [comm_semiring R] (r s : R) :
   (C r : mv_polynomial σ R) = C s ↔ r = s :=
 (C_injective σ R).eq_iff
 
+instance infinite_of_infinite_right (σ : Type*) (R : Type*) [comm_semiring R] [infinite R] :
+  infinite (mv_polynomial σ R) :=
+infinite.of_injective C (C_injective _ _)
+
+instance infinite_of_nonempty_left (σ : Type*) (R : Type*) [nonempty σ] [comm_semiring R] [nontrivial R] :
+  infinite (mv_polynomial σ R) :=
```

Comment:
Do we have `Π (x : σ), polynomial R →+* mv_polynomial σ R` together with a fact that this map is injective? Do we have `instance [comm_semiring R] [nontrivial R] : infinite (polynomial R)`?

Reply:
I added that instance in this PR (in some other file). I don't think we have a name for the map you describe (apart from `eval2 _ _ _`).
If you want I can refactor... but I don't know what would be a good name for the map you describe. (And without a name, it will not make this proof shorter.)

Reply:
If we have no name for this map, then let's forget about this comment.